app/src/main/java/com/example/sharingang/users/FirestoreUserStore.kt

```diff
@@ -63,6 +63,6 @@ class FirestoreUserStore @Inject constructor(private val firestore: FirebaseFire
     override suspend fun hasBeenReported(reporterId: String, reportedId: String): Boolean {
         val docIdRef = firestore.collection("users").document(reportedId)
             .collection("reports").document(reporterId).get().await()
-        return docIdRef.exists()
+        return docIdRef.exists() || reportedId == reporterId
```

Comment:
If the reported id equals the id of the reporter return true? Shouldn't that be an "and"? Maybe I just misunderstood the purpose of this function

Reply:
TL;DR: Actually this change is something I wanted to remove from my last commit because the "or" check is useless.

Its purpose is to return whether the reported has already been reported by the reporter user, therefore if it returns true, then it means that the reporter user does not have the right to report the reported user. In this commit, it also returns true when the user wants to report themself (therefore later on not allowing this action). However, as I have actually moved this check to UserProfileFragment for better understanding, this check becomes useless, and I will remove it.